include/LightGBM/cuda/cuda_utils.h

```diff
@@ -1,24 +1,188 @@
+/*!
+ * Copyright (c) 2021 Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+
 /*!
  * Copyright (c) 2020 IBM Corporation. All rights reserved.
  * Licensed under the MIT License. See LICENSE file in the project root for license information.
  */
+
 #ifndef LIGHTGBM_CUDA_CUDA_UTILS_H_
 #define LIGHTGBM_CUDA_CUDA_UTILS_H_
 
-#ifdef USE_CUDA
-
+#if defined(USE_CUDA) || defined(USE_CUDA_EXP)
 #include <cuda.h>
 #include <cuda_runtime.h>
 #include <stdio.h>
+#include <LightGBM/utils/log.h>
+#endif  // USE_CUDA || USE_CUDA_EXP
+
+#ifdef USE_CUDA_EXP
+#include <vector>
+#endif  // USE_CUDA_EXP
```

Comment:
Why do we need to add `log.h` for old CUDA version? Is it due to added `namespace LightGBM`?
```suggestion
#endif  // USE_CUDA || USE_CUDA_EXP

#ifdef USE_CUDA_EXP
#include <vector>

#include <LightGBM/utils/log.h>
#endif  // USE_CUDA_EXP
```

Reply:
Resolved via e34fcce.